board/opentrons/ot2/rootfs-overlay/bin/ot-machine-id-commit

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env sh
+
+# This script gives the system its persistent machine-id, as part of first-boot
+# setup.
+#
+# /etc/machine-id should contain a persistent, unique identifier for this
+# device. Userspace programs like journald rely on this.
+#
+# On our fresh SD card images, /etc/machine-id is unpopulated. When systemd sees
+# this, it autogenerates a transient machine-id and overlays it, so other
+# userspace programs can rely on its presence.
+#
+# +------------------------------------------+
+# | temp file, populated, created by systemd |
+# +------------------------------------------+
+#                      |
+#                 bind-mounted
+#                      |
+#                      V
+#           +---------------------+
+#           | regular file, empty |
+#           +---------------------+
+#                      |
+#                on filesystem
+#                      |
+#                      V
+#               /etc/machine-id
+#
+# Normally, systemd can, on its own, write back this autogenerated machine-id to
+# persistent storage. (See systemd-machine-id-commit.service.)
+# But in our case, that doesn't work because it expects our active rootfs to be
+# eventually writable, whereas we want to keep it always read-only.
+#
+# This script instead persists the current machine-id by:
+#
+# 1) Mounting our active rootfs in an additional, second place, off to the side
+#    where nothing can accidentally access it. And making that mount writable.
+# 2) Updating the underlying /etc/machine-id file through that second mount.
+#
+# Then, in future boots, the underlying /etc/machine-id file will be populated,
+# and systemd will reuse it instead of autogenerating a new one.
+#
+# See also the update-server service, which needs to apply a similar fixup to
+# system update images.
+
+
+set -eu -o pipefail
+
+MACHINE_ID_FILE=/etc/machine-id
+
+
+# todo(mm, 2021-02-25): Instead of checking like this, should we configure the
+# service to only run on systemd's notion of "first boot"? 
+function already_committed {
+    # Return success if we've already committed the machine-id in a prior boot,
+    # failure otherwise.
+
+    # If $MACHINE_ID_FILE was unpopulated at the beginning of this
+    # boot, then systemd will have bind-mounted a tmpfs file atop it.
+    ! findmnt --mountpoint "$MACHINE_ID_FILE" > /dev/null
+}
+
+function clean_up {
+    umount "$rw_remount_point"
+    rmdir "$rw_remount_point"
+    # todo(mm, 2021-02-25): Should these tolerate failure?
```

Comment:
yes because if this script fails it'll probably set the system to emergency runlevel